src/archive.rs

```diff
@@ -0,0 +1,154 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Tools for creating and inserting into tarballs.
+
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use camino::Utf8Path;
+use flate2::write::GzEncoder;
+use std::convert::TryInto;
+use std::fs::{File, OpenOptions};
+use tar::Builder;
+
+#[async_trait]
+pub trait AsyncAppendFile {
+    async fn append_file_async<P>(&mut self, path: P, file: &mut File) -> std::io::Result<()>
```

Comment:
including `async` in the name here is redundant, it being async is already in the type. normally you see people do this if they have both sync and async variants, but that doesn't look to be the case here.

Reply:
Unfortunately, these *do* have sync variants, see: https://docs.rs/tar/0.4.40/tar/struct.Builder.html . These are the variants that I'm calling internally in the implementation, within the `block_in_place` calls below.

These async methods are basically all extensions of this original interface. I'll add a doc comment to make that more clear.